### DISCUSSION TITLE
page cache profiling with filemap:mm_filemap_add_to_page_cache event

### DISCUSSION THREAD

**Discussion by rajatgoyal247341** [async-profiler/async-profiler#1003] [Q&A] [answered]
I am trying to profile with event as 'filemap:mm_filemap_add_to_page_cache' but it gives below error : 
```
async-profiler-3.0-linux-x64/bin/asprof -e filemap:mm_filemap_add_to_page_cache  -d 10 -f out.html 8368
[ERROR] Unsupported event type
```

Can you help how can I get it.

**Comment by apangin** [comment]
Can you try the [latest build](https://github.com/async-profiler/async-profiler/releases/tag/nightly) from `master`? There were some changes to make this kind of profiling easier.
Also, make sure you run `asprof` under `root`, since kernel tracepoints are typically available only for a privileged user.

**Reply by rajatgoyal247341** [reply]
I downloaded from [latest build](https://github.com/async-profiler/async-profiler/releases/tag/nightly) and ran below with root user. 
```
async-profiler-3.0-268856e5-linux-x64/bin/asprof -e filemap:mm_filemap_add_to_page_cache  -d 10 -f out.html 1
[ERROR] Unsupported event type
```
list command shows only below event types 

```
async-profiler-3.0-268856e5-linux-x64/bin/asprof list 1
Basic events:
  cpu
  alloc
  lock
  wall
  itimer
  ctimer
Java method calls:
  ClassName.methodName
Perf events:
  page-faults
  context-switches
  cycles
  instructions
  cache-references
  cache-misses
  branch-instructions
  branch-misses
  bus-cycles
  L1-dcache-load-misses
  LLC-load-misses
  dTLB-load-misses
  rNNN
  pmu/event-descriptor/
  mem:breakpoint
  trace:tracepoint
  kprobe:func
  uprobe:path

```

**Reply by apangin** [reply]
Does `perf list | grep mm_filemap_add_to_page_cache` show anything?

**Reply by rajatgoyal247341** [reply]
Sorry may be I am missing something, but I don't see any `perf `file [in](https://github.com/async-profiler/async-profiler/releases/download/nightly/async-profiler-3.0-268856e5-linux-x64.tar.gz). 

Below is list of files I see inside tar file
```
find -type f
./lib/libasyncProfiler.so
./LICENSE
./README.md
./bin/asprof
./bin/jfrconv

```

**Reply by apangin** [reply]
It's not from async-profiler, it's a Linux standard tool installable as a regular package.
https://www.man7.org/linux/man-pages/man1/perf.1.html

**Reply by apangin** [reply]
Anyhow, kernel tracepoints may be unavailable if debugfs/tracefs is not mounted.
Check if directories `/sys/kernel/tracing/events` or `/sys/kernel/debug/tracing/events` are non-empty.

If directories are empty, you likely need to mount debugfs pseudo filesystem:
```
mount -t debugfs none /sys/kernel/debug
```

**Reply by apangin** [reply, marked as answer]
It turns out that Linux recently started exposing `tracefs` filesystem at `/sys/kernel/tracing` mount point, comparing to previously used `/sys/kernel/debug/tracing`. Async-profiler looked for tracepoint info only on the latter path. I have updated it to look at both paths.
Try updated [nightly build](https://github.com/async-profiler/async-profiler/releases/tag/nightly) to see if solves the issue.
